python/examples/advanced/Streaming_Data_with_Log_Rotation.ipynb

```diff
@@ -0,0 +1,332 @@
+      "\u001B[0;32m~/Dev/whylogs-v1/python/.venv/lib/python3.9/site-packages/pandas/core/series.py\u001B[0m in \u001B[0;36mname\u001B[0;34m(self)\u001B[0m\n\u001B[1;32m    623\u001B[0m         \u001B[0;34m'Even Numbers'\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m    624\u001B[0m         \"\"\"\n\u001B[0;32m--> 625\u001B[0;31m         \u001B[0;32mreturn\u001B[0m \u001B[0mself\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0m_name\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[0m\u001B[1;32m    626\u001B[0m \u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m    627\u001B[0m     \u001B[0;34m@\u001B[0m\u001B[0mname\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0msetter\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n",
+      "\u001B[0;32m~/Dev/whylogs-v1/python/.venv/lib/python3.9/site-packages/pandas/core/generic.py\u001B[0m in \u001B[0;36m__getattr__\u001B[0;34m(self, name)\u001B[0m\n\u001B[1;32m   5485\u001B[0m         ):\n\u001B[1;32m   5486\u001B[0m             \u001B[0;32mreturn\u001B[0m \u001B[0mself\u001B[0m\u001B[0;34m[\u001B[0m\u001B[0mname\u001B[0m\u001B[0;34m]\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[0;32m-> 5487\u001B[0;31m         \u001B[0;32mreturn\u001B[0m \u001B[0mobject\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0m__getattribute__\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0mself\u001B[0m\u001B[0;34m,\u001B[0m \u001B[0mname\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[0m\u001B[1;32m   5488\u001B[0m \u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m   5489\u001B[0m     \u001B[0;32mdef\u001B[0m \u001B[0m__setattr__\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0mself\u001B[0m\u001B[0;34m,\u001B[0m \u001B[0mname\u001B[0m\u001B[0;34m:\u001B[0m \u001B[0mstr\u001B[0m\u001B[0;34m,\u001B[0m \u001B[0mvalue\u001B[0m\u001B[0;34m)\u001B[0m \u001B[0;34m->\u001B[0m \u001B[0;32mNone\u001B[0m\u001B[0;34m:\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n",
+      "\u001B[0;31mKeyboardInterrupt\u001B[0m: "
+     ]
+    }
+   ],
+   "source": [
+    "url = \"https://blockchain.info/ticker\"\n",
```

Comment:
I disagree with that as having a live data pull allows them to actually see what you might use a data logger on vs the static datasets that I was seeing. Plus in a real world example they would be pulling from something. As a user I would find this much more clear to help figure out how and why to use this than the old logger which pulled a static. 

I'm curious more on your thoughts and on @FelipeAdachi here. :)

Reply:
Ok, so today let's make sure to chat data examples to see if there is a better compromise.

Reply:
Part of my confusion in this scenario (trying to put on my "new person to whylogs hat") is around the poll-and-read-one-message from a URL structure in the example. Having an event driven flow or otherwise varying rate of input over time might better clarify why a rolling logger is well suited for a production use case? The idea of live data is also helpful, which your example shows.

Using the context of a message consumer or web service in an example might both cover the live data idea (without having to actually handle live data) as well as the expectation of varying rates of traffic, needs for scaling characteristics that datasketching is well suited for and focuses on the integration bits. We could simulate sending web traffic or message publishing, a mock function you define in the notebook to produce the data in a mocked way might be fine. In the context of a handle_request method is where the relevant whylogs integration would occur (e.g. given a message and an "ML method" whylogs logger profiles the inputs/outputs with the rolling logger initialized outside of handle_request. Having a rolling logger configured in this case enables the whylogs integration to decouple dataset profile writing from the message flow or web traffic. If two messages come in within one "interval" we get 1 dataset profile written. If we see 1000 messages next interval, still 1 dataset profile written for that interval. The output size of the dataset profiles is relatively predictable even if the traffic might be spiky.

Let's discuss further and chat data sources.

Reply:
Maybe I can have the mock push data at random intervals from 0 to 10 secs or something? Basically mimic a subscription model